File: cpp-lang/ooad/07_gsf/board/tile.cpp
#include "tile.h"

#include <algorithm>
#include <iostream>

void Tile::remove_unit(std::shared_ptr<Unit> unit)
{
    auto it = std::find(units_.begin(), units_.end(), unit);
    if (it == units_.end()) {
        std::cerr << "Cannot find unit to remove" << std::endl;
        return;
    }
    units_.erase(it);
}

File: java-lang/OOAD/Gsf/src/ooad/gsf/board/Tile.java
package ooad.gsf.board;

import java.util.LinkedList;
import java.util.List;

import ooad.gsf.unit.Unit;

public class Tile {

	private List<Unit> units;
	
	public Tile() {
		units = new LinkedList<Unit>();
	}
	
	protected void addUnit(Unit unit) {
		units.add(unit);
	}
	
	protected void removeUnit(Unit unit) {
		units.remove(unit);
	}
	
	protected void removeUnits() {
		units.clear();
	}
	
	protected List<Unit> getUnits() {
		return units;
	}
}
